core/Archive/ArchiveInvalidator.php

```diff
@@ -654,6 +654,10 @@ public function removeInvalidationsFromDistributedList($idSites, $pluginName = n
         $list = new ReArchiveList();
         $entries = $list->getAll();
 
+        // Make sure that idSites is an array or all before using it
+        $idSites = empty($idSites) || 'all' === $idSites
```

Comment:
Shouldn't this be done in the method that is calling this one? This methods seems only be used by `removeInvalidationsSafely`, which also calls `removeInvalidations`. The latter one might also have problems if an unexpected value is passed in...

Reply:
@sgiehl I thought about that, but `removeInvalidations` calls Model, which uses `getRemoveInvalidationsIdSitesClause` to handle whether idSites is an int or array.

Reply:
That's a good point. I just updated the fix to only automatically fix when an int is passed.